Library/Homebrew/utils/pypi.rb

```diff
@@ -212,12 +212,13 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
       end
     end
 
-    ensure_formula_installed!("pipgrip")
+    ensure_formula_installed!("python")
 
     ohai "Retrieving PyPI dependencies for \"#{input_packages.join(" ")}\"..." if !print_only && !silent
     command =
-      [Formula["pipgrip"].opt_bin/"pipgrip", "--json", "--tree", "--no-cache-dir", *input_packages.map(&:to_s)]
-    pipgrip_output = Utils.popen_read(*command)
+      [Formula["python"].bin/"python3", "-m", "pip", "install", "-q", "--dry-run", "--ignore-installed", "--report",
+       "/dev/stdout", *input_packages.map(&:to_s)]
+    pip_output = Utils.popen_read({ "PIP_REQUIRE_VIRTUALENV" => "false" }, *command)
```

Comment:
NB: I've added `PIP_REQUIRE_VIRTUALENV` here to override the user's `pip` `require-virtualenv` setting, if present. Without this, a user who has `require-virtualenv` set will silently break this developer command, since `require-virtualenv` doesn't currently respect the `--dry-run` flag.